### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by iBel251**
- Create a class `Nameable`.
   - Implement a method called `correct_name` that will raise a `NotImplementedError`.
- Turn Person class to Nameable.
   - Make sure that the Person class inherits from Nameable
   - Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.
- Prepare base Decorator
   - Make sure that it inherits from Nameable.
   - In the constructor assign a nameable object from params to an instance variable.
   - Implement the `correct_name` method that returns the result of the `correct_name` method of the `@nameable`
- Prepare CapitalizeDecorator and TrimmerDecorator.

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Nameable
+  def initialize(nameable)
+    @nameable = nameable
+  end
+
+  attr_accessor :nameable
```
Comment (BenMukebo) [inline nameable.rb:6]: - [ ] I don't see the `Decorator` class which must inherit from the `Namable` class. 🤔  kindly add this and make sure that:
  - It inherits from Nameable.
  - In the constructor, a namable object from params is assigned to an instance variable.
  - You have implemented the correct_name method that returns the result of the correct_name method of the @nameable.

- The `Decorator` class, might look similar to:

```ruby
class Decorator < Nameable
    attr_accessor :nameable
  
    def initialize(nameable)
        ....
    end
  

    def correct_name
           ....
    end
end
```

File: nameable.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
```
Comment (BenMukebo) [inline nameable.rb:7]: - [ ] I also don't see the `CapitalizeDecorator` class. 🤔 kindly, Create a class that inherits from the base Decorator class. and make sure that you Implement a method correct_name that capitalizes the output of @nameable.correct_name.

- The `CapitalizeDecorator` class, might look similar to:

```ruby
# Inherits from Decorator
class CapitalizeDecorator < Decorator
    # Capitalize the first letter of the given name.
    def correct_name
      @nameable.correct_name.capitalize
    end
end
```

- [ ] I also don't see the `TrimmerDecorator` class. 🤔 kindly, Create a class that inherits from the base Decorator class. Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

- The  `TrimmerDecorator` class, might look similar to:

```ruby
# Inherits from Decorator
class TrimmerDecorator < Decorator
    # The name has a maximum of 10 characters. If it's longer it should trim the word.
    def correct_name
       ....
    end
end
```

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @iBel251,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: nameable.rb (hunk 2 above)
Comment (iBel251) [reply nameable.rb:7]: Hi thank you the review :)
All the decorator classes are inside the `base.rb` file.

Comment (BenMukebo) [reply nameable.rb:7]: Hi @iBel251, I can now see that you implemented the  `CapitalizeDecorator` class and the `TrimmerDecorator` class inside the  `base.rb` file. Great job 👏🏻  however, your code does not work as expected see the screenshot 👇🏻 




- As you can see on line 37, The name has a maximum of 10 characters. If it's longer it should trim the word.

Comment (iBel251) [reply nameable.rb:7]: oww yes i see it now, I used `capitalize` instead of `upcase` keyword. I'll fix it now.

**Pull request merged**